frontend/src/main/components/Commons/ManageCows.js

```diff
@@ -1,25 +1,28 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { Card, Button, Row, Col, Modal, Form } from "react-bootstrap";
 import cowHead from "./../../../assets/CowHead.png";
-// add parameters 
+
 const ManageCows = ({ userCommons, commons, onBuy, onSell }) => {
-  // Stryker disable next-line all : hard to set up test for caching
   const [showBuyHerdModal, setShowBuyHerdModal] = useState(false);
   const [numOfCowsToBuy, setNumOfCowsToBuy] = useState(2);
+  const [herdCost, setHerdCost] = useState(0);
 
-  //  handlers for the  herd modal and the number of cows to buy
-  // Stryker disable next-line all
   const handleBuyHerdModalClose = () => setShowBuyHerdModal(false);
   const handleBuyHerdModalShow = () => setShowBuyHerdModal(true);
+
   const handleNumOfCowsToBuyChange = (event) =>
     setNumOfCowsToBuy(parseInt(event.target.value));
 
   const handleBuyHerd = () => {
-    // Stryker disable next-line all
     onBuy({ ...userCommons, numOfCows: numOfCowsToBuy });
     handleBuyHerdModalClose();
   };
-
+// Stryker disable all
+  useEffect(() => {
```

Comment:
We don’t need an effect here. This can just be a variable.

Reply:
By extension the herd cost doesn’t need to be a state.

Reply:
Why not use an effect? It adjusts the price whenever the user changes the num of cows on the buy herd modal

Reply:
The effect here will calculate cost after the component renders, which triggers another render to sync the cost with its dependencies. This is not ideal because it triggers an additional rerender and the cost and its dependencies will be out of sync until the following render. The best way to do this is to calculate during render by bringing `const cost = numOfCowsToBuy * commons?.cowPrice;` out of the effect and remove the state.